pkg/store/etcd.go

```diff
@@ -99,13 +86,21 @@ func (s *Etcd) Get(key string) (*KVPair, error) {
 }
 
 // Put a value at "key"
-func (s *Etcd) Put(key string, value []byte) error {
-	if _, err := s.client.Set(key, string(value), 0); err != nil {
+func (s *Etcd) Put(key string, value []byte, opts *RequestOptions) error {
+
+	// Default TTL = 0 means no expiration
+	ttl := 0 * time.Second
+	if opts != nil && opts.Ephemeral {
+		ttl = 60 * time.Second
+		go s.periodicRenew(key, value, ttl)
```

Comment:
Unfortunately, I think we have to keep track of our goroutines :/

This means that every time we do an ephemeral put of the same key, we end up with a new goroutine.

We could use a kind of "set" (`map[string]struct{}`) and conditionally spawn goroutines